File: MATTHEW/leetcode/AIcoding/4.GPT-4o/java_solutions/2735-collecting-chocolates.java
class Solution {
    public int maxChoco(int[] chocolates, int K) {
        int n = chocolates.length;
        if (n == 0 || K <= 0) return 0;

        Arrays.sort(chocolates);
        int maxChoco = 0;

        for (int i = n - 1; i >= 0 && K > 0; i--) {
            if (chocolates[i] <= K) {
                maxChoco += chocolates[i];
                K -= chocolates[i];
            } else {
                maxChoco += K;
                break;
            }
        }

        return maxChoco;
    }
}

File: leetcode/AIcoding/4.GPT-4o/c_solutions/2735-collecting-chocolates.c
int maxChocolates(int* chocolates, int chocolatesSize, int k) {
    int maxChoco = 0;
    for (int i = 0; i < chocolatesSize; i++) {
        if (i + k < chocolatesSize) {
            int total = 0;
            for (int j = i; j < i + k; j++) {
                total += chocolates[j];
            }
            if (total > maxChoco) {
                maxChoco = total;
            }
        }
    }
    return maxChoco;
}